### ISSUE TITLE
Questing areas and talking

### ISSUE THREAD

**Issue by ghost** [Explv/Explvs-AIO#43]
So, if a quest has an area where the bot is supposed to go and talk to someone, it will go the area and try to find the person. If the person is out of the specified area it, then it will go and talk to the npc but then realize it's not in the area anymore and try to go to the area specified, therefore getting stuck in a look of going back and forth, should be a simple check if the bot is already talking to the npc.

**Comment by Azn2000** [comment]
In a task that i've wrote and currently testing im not checking for the area im checking for the npc found maybe this also works for your problem

```java
NPC target = getNpcs().closest("Some NPC name");

if(target != null){
    if (!getDialogues().inDialogue() || !myPlayer().isInteracting(target)) {
        if (target.interact("Talk-to")) {
            Sleep.sleepUntil(() -> getDialogues().inDialogue() && myPlayer().isInteracting(target), 5000);
        }
    } else {
        getDialogues().completeDialogue(DIALOG_OPTIONS);
    }
}else{
    //Move to the area
    getWalking().webWalk(target);
}
```

**Comment by Azn2000** [comment]
I dont know how usefull this is to you because of more building levels etc, needs probably some changes.

**Comment by RealPoke** [comment]
#46 should fix this :)

**Comment by RealPoke** [comment]
@Azn2000 what happened to this fix?
I am looking to fix this, but was wondering if you already have a fix that you closed?! :o

**Comment by Explv** [comment]
Fixed in d79cfd5306ada1b301fde81ab4581baf60f97f13
